app/user/profile/LogoutButton.tsx

```diff
@@ -3,9 +3,16 @@
 import { useRouter } from "next/navigation"
 
 import { Button } from "@/components/shared/button/Button"
+import useSession from "@/hooks/useSession"
 
-const UniloginLogoutButton = () => {
+const LogoutButton = () => {
+  const session = useSession()
   const router = useRouter()
+
+  if (!session?.session?.isLoggedIn) {
+    return null
+  }
+
   return (
     <>
       <Button onClick={() => router.push("/auth/logout")} ariaLabel="Log ud">
```

Comment:
In order to prevent similar problems as we had in the CMS regarding the logout button and session, try to move this callback function out of here into something like `const handleOnClick = () => {...}` , add a state to the component that tracks whether it's already been clicked or not. Then we can tie that state to `isDisabled` on the `<Button />` in order to prevent multiple clicks in short succession on that button.